### PR TITLE
Add CropAndResize gradients for C++ image gradient operators

### PR DESCRIPTION
**Pull request by a6802739**
Add gradient and test for `CropAndResize`, and just refer to the implementation of `image_grad.py`.

But something still goes wrong, when I run the uint test, it will notice me that `Not found: No gradient defined for op: CropAndResize`.

I hope I could get some reference in the thread #21019, but no response there, so I come up with the PR and hope someone could give me some reference.

Thanks.

### CODE REVIEW

**Comment by skye** [comment]
You need to use the REGISTER_GRADIENT_OP macro. There are examples of it in image_grad.cc. It should be something like `REGISTER_GRADIENT_OP("CropAndResize", CropAndResizeGradHelper);`

**Comment by a6802739** [comment]
@skye,  Thanks for your guidance, I have updated it now and pass the test. 

Please have a look at it. Thanks.

File: tensorflow/cc/gradients/image_grad.cc [hunk 1]
```diff
@@ -86,6 +86,27 @@ Status ScaleAndTranslateGradHelper(const Scope& scope, const Operation& op,
...
 
+Status CropAndResizeGradHelper(const Scope& scope, const Operation& op,
+                                   const std::vector<Output>& grad_inputs,
+                                   std::vector<Output>* grad_outputs) {
+  DataType input_type;
+  string method;
+  TF_RETURN_IF_ERROR(
+      GetNodeAttr(op.node()->attrs(), "method", &method));
+  TF_RETURN_IF_ERROR(
+      GetNodeAttr(op.node()->attrs(), "T", &input_type));
```
Comment (skye) [inline tensorflow/cc/gradients/image_grad.cc]: Should you check for allowed types like in the Python version? https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/image_grad.py#L107

**Review by skye — CHANGES_REQUESTED** [review]

File: tensorflow/cc/gradients/image_grad.cc (hunk 1 above)
Comment (a6802739) [reply tensorflow/cc/gradients/image_grad.cc]: I think it'll check the type during we build the code, so it has no need to check the types in the c plus plus version.

**Comment by skye** [comment]
Looks good, except I think there are some minor formatting errors. Can you run `clang-format -style=google` over the files?

**Comment by a6802739** [comment]
@skye , I have formatted the code as you say, and thanks for your review.

**Review by skye — APPROVED** [review]
Looks great, thanks for contributing!

**Comment by a6802739** [comment]
@skye, Thanks for your help. It seems some CI build  failed, when I see the details, some irrelevant test failed info occured.  Just like some info within `Ubuntu Python3`:
```
ERROR: /tmpfs/src/github/tensorflow/tensorflow/python/kernel_tests/linalg/BUILD:201:1: Couldn't build file tensorflow/python/kernel_tests/linalg/linear_operator_inversion_test/shard_1_of_5/test.log:  failed (Exit 34). Note: Remote connection/protocol failed with: execution failed Call failed with not retriable error: io.grpc.StatusRuntimeException: PERMISSION_DENIED: The caller does not have permission: 
```

How could I fix these?

**Comment by skye** [comment]
Let's try rerunning the tests...

**Pull request merged**